tfjs-backend-webgl/src/shader_compiler.ts

```diff
@@ -323,7 +323,17 @@ function getShaderPrefix(glsl: GLSL): string {
     };
 
     uniform float NAN;
-    ${glsl.defineSpecialNaN}
+    // Use custom isnan definition to work across differences between
+    // implementations on various platforms. While this should happen in ANGLE
+    // we still see differences between android and windows (on chrome) when
+    // using isnan directly.
+    #define isnan(value) isnan_custom(value)
+    bool isnan_custom(float val) {
+      return val != val || !(val > 0. || val < 1. || val == 0.0);
```

Comment:
The previous versions for WebGL 1 and 2 are slightly different. Why is it "val < 1." instead of "val < 0."? A typo?